.github/workflows/knn-index.yml

```diff
@@ -0,0 +1,72 @@
+name: Knn tests
+# This workflow is triggered on pull requests to main branch
+on:
+  pull_request:
+    branches:
+      - '*'
+  push:
+    branches:
+      - '*'
+
+jobs:
+  req:
+    # Job name
```

Comment:
Let's not add a separate workflow. Instead, try to install knn plugin always and skip the test if plugin is unavailable. 

Although this can lead to silent failures if we're always failing to install knn plugin due to some bug. Just thinking out loud, we can probably enforce that non-snapshot builds has to run the knn tests.

Reply:
Used same security workflow.